marimo/_plugins/ui/_impl/plotly_plot.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2023 Marimo. All rights reserved.
+from __future__ import annotations
+
+import json
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    Dict,
+    Final,
+    List,
+    Optional,
+)
+
+from marimo import _loggers
+from marimo._dependencies.dependencies import DependencyManager
+from marimo._output.rich_help import mddoc
+from marimo._plugins.core.web_component import JSONType
+from marimo._plugins.ui._core.ui_element import UIElement
+
+LOGGER = _loggers.marimo_logger()
+
+if TYPE_CHECKING:
+    import plotly.graph_objects as go  # type:ignore
+
+# Selection is a dictionary of the form:
+# {
+#   "points": {
+#     "field1": "value1",
+#     "field2": "value2",
+#   }[],
+#   "range": {
+#     "field1": [min, max],
+#     "field2": [min, max],
+#   },
+# }
+PlotlySelection = Dict[str, JSONType]
+
+
+@mddoc
+class plotly_plot(UIElement[PlotlySelection, List[Dict[str, Any]]]):
+    """Make reactive plots with Plotly Express.
+
+    Use `mo.ui.plotly_plot` to make plotly plots reactive: select data
+    with your cursor on the frontend, get them as a list of dicts in Python!
+
+    **Example.**
+
+    ```python
+    import plotly.express as px
+    import marimo as mo
+    from vega_datasets import data
+
+    _plot = px.scatter(
+        data.cars(),
+        x="Horsepower",
+        y="Miles_per_Gallon",
+        color="Origin"
+    )
+
+    plot = mo.ui.plotly_plot(_plot)
+    ```
+
+    ```
+    # View the plot and selected data
+    mo.hstack([plot, plot.value])
+    ```
+
+    **Attributes.**
+
+    - `value`: a dict of the plot data
```

Comment:
Are we returning as a `dict` instead of a Pandas dataframe because Pandas is an optional dependency?